### PR TITLE
Pynecone App should prevent all pycache folders and all .pyc, .pyo, .pyd files from being uploaded.

### PR DESCRIPTION
**Pull request by milochen0418**
Pynecone. App should prevent all pycache folders and all .pyc, .pyo, .pyd files from being uploaded.

### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### New Feature Submission:

- [x] Does your submission pass the tests? 

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

`pc init` will create the .gitignore  file that cannot prevent `__pycache__` upload to the git. 
And python application developer need this function.

### CODE REVIEW

File: pynecone/constants.py
Comment (ElijahAhianyo) [inline pynecone/constants.py:139]: we should probably add these to the .gitignore file instead

**Review by ElijahAhianyo — CHANGES_REQUESTED** [review]

File: pynecone/constants.py
Comment (danik292) [reply pynecone/constants.py:139]: What .gitignore file is?

Comment (milochen0418) [reply pynecone/constants.py:139]: In the view of the application developers,  they use high-level to create the basic project by using ```pc init```. Because the .gitignore is contained in the basic project,  they often assume this .gitignore is enough to prevent uploading the building things to the git server.

For the python project, .gitignore often contain  "__pycache__/", "*.py[cod]". 
And pynecone's web application is a python project too. 

__pycache__/, .pyc, .pyo, .pyd are all compiled byte code
Python programmers usually ignore bytecode.

Comment (danik292) [reply pynecone/constants.py:139]: ok

Comment (chrisrosner) [reply pynecone/constants.py:139]: > we should probably add these to the .gitignore file instead

The .gitignore file in pinecone is distinct from the .gitignore that is created by pc using these constants. This seems like the right approach, and a change that seems like it would benefit everyone, @ElijahAhianyo

**Review by danik292 — APPROVED** [review]

File: pynecone/constants.py
Comment (milochen0418) [reply pynecone/constants.py:139]: Or we may need some reason why Pynecone project needs to keep `__pycache__` on git. 
If there is some reason to keep `__pycache__` on pynecone project, then we can review the following PR.
- https://github.com/pynecone-io/pynecone/pull/1094

Two PR are  okay for me, because they at least give one way for the developer who want to push pure source code on git.

**Review by chrisrosner — APPROVED** [review]

File: pynecone/constants.py
Comment (milochen0418) [reply pynecone/constants.py:139]: @ElijahAhianyo  Can you give one reason about why git should keep `__pycache__` ? 
Because if you don't approve this PR and don't give reason. 
@chrisrosner  will keep to reject this PR 
- https://github.com/pynecone-io/pynecone/pull/1094

You keep to reject this PR, and chrisrosner keep to reject another PR ... 
Then  it must be a disaster for all pynecone app developer.

**Review by TaiJuWu — APPROVED** [review]

**Comment by milochen0418** [comment]
After communication, we have the summmary.  
--
In the discord #opensource channel, there is a discuss about this PR. 
Because there are at least 3 .gitignore files in the pynecone app project 
- app's root
- app (python part named as app_name you set in pc init)
- web (web part   named as .web)
After discussion and understanding, I explain what happened in the discord channel. 

.gitignore will influent each other .gitignore. The core team needs to think about how to manage them to get the most flexibility first. And then we can start to think what to do next.  
In the future, the solution maybe is the copying some .gitignore file to the app's root, or support some other thing to solve.  Or some other way else.

**Review by picklelo — APPROVED** [review]

**Pull request merged**
